### PR TITLE
[LTC] Implement The Balrog of Moria

### PR DESCRIPTION
**Pull request by Susucre**
There is still something wrong with the multi targetting in multiplayer environment. For now I suspect the ExileTargetEffect() to not expect multi targetting.

Seems to work in 1v1.

### CODE REVIEW

**Comment by Susucre** [comment]
[[The Balrog of Moria]]

File: Mage.Sets/src/mage/cards/t/TheBalrogOfMoria.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Haste
+        this.addAbility(HasteAbility.getInstance());
+
+        // When The Balrog of Moria dies, you may exile it. When you do, for each opponent, exile up to one target creature that player controls.
+        ReflexiveTriggeredAbility reflexiveAbility = new ReflexiveTriggeredAbility(
+            new ExileTargetEffect()
+                .setTargetPointer(new EachTargetPointer())
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/t/TheBalrogOfMoria.java:57]: Something is weird with `ExileTargetEffect()` and that manually set `TargetPointer`.

When testing the card with multiple opponent, each with a valid creature to target,
the ability is properly asking for a target per opponent.

However in ExileTargetExile.java, we have the following loop over the targets:

```@Override
    public boolean apply(Game game, Ability source) {
        Player controller = game.getPlayer(source.getControllerId());
        if (controller != null) {
            Set<Card> toExile = new LinkedHashSet<>();
            for (UUID targetId : getTargetPointer().getTargets(game, source)) {```

That `getTargetPointer()` is not the `EachTargetPointer()` I was expecting (it is a `FirstTargetPointer` with a single target). `source.getTargets()` has all the targetted creatures under it.

I am not sure what is wrong there, is it a bug where a wrong `TargetPointer` is copied at some point, or is the way I did set up the targetting wrong in the balrog's code?

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/t/TheBalrogOfMoria.java:57]: My guess is that `.setTargetPointer(new EachTargetPointer())` has to be on the `DoWhenCostPaid` which copies its target pointer to the reflexive ability

Comment (Susucre) [reply Mage.Sets/src/mage/cards/t/TheBalrogOfMoria.java:57]: thanks, I'll test that instead.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/t/TheBalrogOfMoria.java]: That was it. Now the multiple targets are properly exiled.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**